### PR TITLE
Catalogue repository and descriptor repository ports

### PR DESCRIPTION
**Pull request by leynos**
## Summary by Sourcery

Add read-side catalogue and descriptor repository ports with Diesel-backed adapters, shared JSON decode helpers, and behavioural tests, while improving embedded Postgres test orchestration and documenting the completed roadmap item.

New Features:
- Introduce CatalogueRepository and DescriptorRepository domain ports with cohesive snapshot types for explore catalogue and descriptor registries.
- Provide DieselCatalogueRepository and DieselDescriptorRepository implementations that assemble validated domain snapshots from PostgreSQL using deterministic ordering.
- Add JSONB-to-domain decode helpers for localization maps, image assets, and semantic icon identifiers used by read repositories.

Enhancements:
- Refine backend architecture documentation with design decisions for catalogue and descriptor read ports and snapshot handling.
- Add an execution plan document describing the implementation of catalogue and descriptor repository ports.
- Expose read-model Diesel row types separately from ingestion models for clearer persistence boundaries.

Build:
- Split the Makefile test target into separate Rust and frontend test targets and prevent Rust nextest runs from failing fast.

CI:
- Adjust nextest configuration to serialize embedded Postgres test binaries via a pg-embed test group and shared cluster, improving stability under parallel runs.

Documentation:
- Update the backend roadmap to mark catalogue and descriptor repository ports as complete.

Tests:
- Introduce BDD-style behaviour tests for catalogue and descriptor read repositories using embedded PostgreSQL and snapshot assertions.
- Add unit tests for JSON decode helpers covering round-trip, validation failure, and edge cases.
- Refactor existing embedded Postgres tests to use a shared_cluster_handle helper and centralise process-exit cleanup to avoid cross-binary interference.

Chores:
- Add markdown-it as a dev dependency for tooling scripts.

### CODE REVIEW

File: backend/Cargo.toml [hunk 1]
```diff
@@ -78,12 +78,12 @@ actix-codec = "0.5.2"
 # Actix Web Client for WebSocket and HTTP integration testing.
 awc = "3.8.1"
 env-lock = "1.0.1"
-libc = "0.2"
 futures = "0.3"
 rstest = "0.26"
 rstest-bdd = "0.5.0"
 rstest-bdd-macros = { version = "0.5.0", features = ["strict-compile-time-validation"] }
 insta = { version = "1", features = ["json", "redactions"] }
+libc = "0.2"
```
[the remark this answers is not in the document]
Comment (leynos) [reply backend/Cargo.toml]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.

**Comment by leynos** [comment]
@coderabbitai Please suggest a fix for this issue and supply a prompt for an AI coding agent to enable it to apply the fix:

## Code Duplication
backend/tests/catalogue_descriptor_read_models_bdd.rs: 

### What lead to degradation?
The module contains 6 functions with similar structure: a_malformed_descriptor_localization_row_is_inserted_directly,a_malformed_localization_row_is_inserted_directly,get_catalogue_snapshot,get_descriptor_snapshot and 2 more functions

### Why does this problem occur?
Duplicated code often leads to code that's harder to change since the same logical change has to be done in multiple functions. More duplication gives lower code health.

### How to fix it?
A certain degree of duplicated code might be acceptable. The problems start when it is the same behavior that is duplicated across the functions in the module, ie. a violation of the Don't Repeat Yourself (DRY) principle. DRY violations lead to code that is changed together in predictable patterns, which is both expensive and risky. DRY violations can be identified using CodeScene's X-Ray analysis to detect clusters of change coupled functions with high code similarity. [Read More](https://codescene.com/blog/software-revolution-part3/)
Once you have identified the similarities across functions, look to extract and encapsulate the concept that varies into its own function(s). These shared abstractions can then be re-used, which minimizes the amount of duplication and simplifies change.

File: backend/src/outbound/persistence/diesel_catalogue_repository.rs [hunk 2]
```diff
@@ -0,0 +1,308 @@
...
+            }
+            .scope_boxed()
+        })
+        .await
+        .map_err(map_diesel_error)
+    }
+
+    fn convert_catalogue_rows(
+        rows: CatalogueRows,
+    ) -> Result<ConvertedCatalogueRows, CatalogueRepositoryError> {
```
[the remark this answers is not in the document]
Comment (leynos) [reply backend/src/outbound/persistence/diesel_catalogue_repository.rs]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.

File: backend/src/outbound/persistence/json_serializers.rs [hunk 3]
```diff
@@ -31,3 +40,166 @@ pub(super) fn image_asset_to_json(image: &ImageAsset) -> Value {
...
+///
+/// ```rust,ignore
+/// let json = serde_json::json!({"en-GB": {"name": "Scenic"}});
+/// let map = json_to_localization_map(&json).unwrap();
+/// assert!(map.as_map().contains_key("en-GB"));
+/// ```
+pub(super) fn json_to_localization_map(value: &Value) -> Result<LocalizationMap, String> {
+    let raw: BTreeMap<LocaleCode, LocalizedStringSet> =
+        serde_json::from_value(value.clone()).map_err(|e| format!("localization decode: {e}"))?;
+    LocalizationMap::new(raw).map_err(|e| format!("localization validation: {e}"))
```
[the remark this answers is not in the document]
Comment (leynos) [reply backend/src/outbound/persistence/json_serializers.rs]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.
